.github/workflows/molecule-firewalld.yml

```diff
@@ -0,0 +1,39 @@
+name: Test firewalld
+on:
+  pull_request:
+    paths:
+      - "roles/firewalld/**"
+
+jobs:
+  molecule-firewalld:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        scenario:
+          - centos7
+    env:
+      MOLECULE_RUN_TAGS: firewalld
+      PY_COLORS: 1
+      ANSIBLE_FORCE_COLOR: 1
+    steps:
+      - name: Check out the codebase
+        uses: actions/checkout@v4
+        with:
+          path: ansible_collections/mirsg/infrastructure
```

Comment:
we should update our `molecule-test` action to take the checkout path as input so we don't duplicate workflows